src/pages/todo/sagas.js

```diff
@@ -1,9 +1,15 @@
 import { watchApi } from '../../common/api'
+import { fork } from 'redux-saga/effects'
 
 import {
-  LIST_TODOS
+  LIST_TODOS,
+  DETAIL_TODOS
 } from './actions';
 
 export default function* todoSagas() {
-  yield* watchApi(LIST_TODOS, '/v1/todos')
+  yield [
+    fork(watchApi, LIST_TODOS, '/v1/todos'),
```

Comment:
@vinhkq '/v1/todos/' should be '/v1/todos/${id}'
and then in common/api.js:
```
if(dispatch.id) {
  path = path + dispatch.id
}
```
should be
```
path = format(path, action)
(where `action` is renamed from `dispatch` )
```
with defining a format function:
```
function format(template, replacement)
{
    if (typeof replacement != "object")
    {
        replacement = Array.prototype.slice.call(arguments, 1);
    }
    return template.replace(/\${(.+?)\}/g, function(m, c)
    {
        return (replacement[c] != null) ? replacement[c] : m
    });
}
```